Parse/PFObject.m

```diff
@@ -2006,6 +2006,33 @@ - (void)deleteInBackgroundWithBlock:(PFBooleanResultBlock)block {
     [[self deleteInBackground] thenCallBackOnMainThreadWithBoolValueAsync:block];
 }
 
+- (void)revert {
+    @synchronized (lock) {
+        if ([self isDirty]) {
+            [operationSetQueue removeAllObjects];
```

Comment:
This should remove all the objects from the `unsavedChanges`, rather than the entire operationSetQueue, since the save operation could be in-flight right now and we don't want to revert that.